libafl/src/feedbacks/map.rs

```diff
@@ -388,24 +393,25 @@ pub struct MapFeedback<N, O, R, S, T> {
     /// Name of the feedback as shown in the `UserStats`
     stats_name: String,
     /// Phantom Data of Reducer
-    phantom: PhantomData<(N, O, R, S, T)>,
+    phantom: PhantomData<(C, N, O, R, S, T)>,
 }
 
-impl<N, O, R, S, T> UsesObserver<S> for MapFeedback<N, O, R, S, T>
+impl<C, N, O, R, S, T> UsesObserver<S> for MapFeedback<C, N, O, R, S, T>
 where
     S: UsesInput,
-    O: Observer<S>,
+    C: AsRef<O> + Observer<S>,
```

Comment:
@addisoncrump This guy cannot just reuse the `O` generic?

Reply:
No. The reason I named it `A` is because `AsRef<O>` is used as a generalisation over the map observer.

If we are using a tracking hint, then `A = TrackingHinted<O>`. If not, `A = O`. This allows us to encode type hints, then use `AsRef` to access the actual map in the background.